modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java

```diff
@@ -63,29 +63,77 @@ public String evaluate(SQLContext context) throws EdmException {
         if (isEmpty()){
             return "";
         }
-        boolean caseSensitive = Boolean.parseBoolean(Configuration.get("DIRIGIBLE_DATABASE_NAMES_CASE_SENSITIVE", "false"));
 
         StringBuilder join = new StringBuilder();
+
+        boolean hasFirstJsonMappingTable = query.hasSQLMappingTablePresent(start, target);
+        boolean hasSecondJsonMappingTable = query.hasSQLMappingTablePresent(target, start);
+
+        if(hasFirstJsonMappingTable && hasSecondJsonMappingTable) {
```

Comment:
What will happen if mapping table from the two entities is different? Can we add check for that?

Reply:
Resolved in [f66d649](https://github.com/eclipse/dirigible/pull/1685/commits/f66d649aeb8e56d59d0b4a748b56a5d89dd8c1fd)